google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/BuildDeploymentSourceType.java

```diff
@@ -41,7 +41,12 @@
  * for specific build-systems should extend this providing the build-system specific
  * implementations.
  */
-public abstract class BuildDeploymentSourceType extends ModuleDeploymentSourceType {
+public abstract class BuildDeploymentSourceType
+    extends DeploymentSourceType<ModuleDeploymentSource> {
```

Comment:
In case anyone was wondering why I changed it to extend DeploymentSourceType instead of ModuleDeploymentSourceType: in order to deserialize our maven sources, we need to be able to set an ID (this is how the platform locates the object to load). ModuleDeploymentSourceType does not expose a constructor to set this id (and wasn't adding any useful functionality on top of DeploymentSourceType.